pkg/repository/repository.go

```diff
@@ -77,9 +78,30 @@ func (r RepositoryImpl) ImageMetadata(imgRef string, insecure bool) (imageInfo *
 	return imageInfo, nil
 }
 
-func (r RepositoryImpl) PushImage(img v1.Image, imgRef string) error {
-	if err := crane.Push(img, imgRef, crane.WithContext(context.Background())); err != nil {
-		return fmt.Errorf("error pushing image %q: %v", img, err)
+func (r RepositoryImpl) PullImage(ctx context.Context, imgRef string, insecure bool) (v1.Image, error) {
```

Comment:
Do you think it makes sense to just use https://github.com/google/go-containerregistry/blob/main/pkg/crane/copy.go for copying? I think it just streams the image to the remote locatoin with a pre-pull.

Reply:
Yes, of course. I missed that somehow.